tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,318 @@
+    expect(root.find('.UserProfile-number-of-addons').children())
+      .toHaveText('70');
+  });
+
+  it('renders LoadingText for number of add-ons while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-number-of-addons')).toHaveLength(1);
+    expect(root.find('.UserProfile-number-of-addons').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's average add-on rating", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        average_addon_rating: 4.1,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(Rating))
+      .toHaveProp('rating', 4.1);
+  });
+
+  it('renders LoadingText for average add-on rating while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-tofu' } });
+
+    expect(root.find('.UserProfile-rating-average')).toHaveLength(1);
+    expect(root.find('.UserProfile-rating-average').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's biography", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: 'Not even vegan!',
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography').render().html())
+      .toContain('Not even vegan!');
+  });
+
+  it('omits a null biography', () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: null,
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography')).toHaveLength(0);
+  });
+
+  it('omits an empty biography', () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        biography: '',
+        username: 'tofumatt',
+      },
+    });
+    const root = renderUserProfile({ store });
+
+    expect(root.find('.UserProfile-biography')).toHaveLength(0);
+  });
+
+  it('renders a report abuse button if user is loaded', () => {
+    const root = renderUserProfile();
+
+    expect(root.find(ReportUserAbuse)).toHaveLength(1);
+  });
+
+  it('still renders a report abuse component if user is not loaded', () => {
+    // The ReportUserAbuse handles an empty `user` object so we should
+    // always pass the `user` prop to it.
+    const root = renderUserProfile({ params: { username: 'not-loaded' } });
+
+    expect(root.find(ReportUserAbuse)).toHaveLength(1);
+  });
+
+  it('renders a not found page if the API request is a 404', () => {
+    const { store } = dispatchSignInActions();
+    const errorHandler = new ErrorHandler({
+      id: 'some-id',
+      capturedError: { responseStatusCode: 404 },
```

Comment:
This looks pretty fragile. [Here](https://github.com/mozilla/addons-frontend/blob/ec3a4dd837ac92516283f3db53e3c21aaec6efc8/tests/unit/amo/components/TestCollection.js#L485-L493) is a better way to do it.

Reply:
I feel like I tried that and something broke, but I was probably doing it wrong™.

Reply:
It worked, I think I must've just forgot to pass `store.dispatch` last time 😓